compiler/luci-interpreter/src/kernels/UnidirectionalSequenceLSTM.cpp

```diff
@@ -88,12 +88,251 @@ void UnidirectionalSequenceLSTM::check_input_tensor_dimensions(int n_input, int
                                                                int n_cell, bool use_layer_norm,
                                                                bool is_integer)
 {
-  // TODO implement
-  (void)n_input;
-  (void)n_output;
-  (void)n_cell;
-  (void)use_layer_norm;
-  (void)is_integer;
+  // Making sure clipping parameters have valid values.
+  // == 0 means no clipping
+  //  > 0 means clipping
+  LUCI_INTERPRETER_CHECK(params().cell_clip >= 0);
+  LUCI_INTERPRETER_CHECK(params().proj_clip >= 0);
+
+  if (input_to_input_weights() != nullptr)
+  {
+    const Shape &input_to_input_weights_shape = input_to_input_weights()->shape();
+    LUCI_INTERPRETER_CHECK(input_to_input_weights_shape.num_dims() == 2);
+    LUCI_INTERPRETER_CHECK(input_to_input_weights_shape.dim(0) == n_cell);
+    LUCI_INTERPRETER_CHECK(input_to_input_weights_shape.dim(1) == n_input);
```

Comment:
Original TF code has type check

```
    TF_LITE_ENSURE_TYPES_EQ(context, input_to_input_weights->type,
                            input_to_forget_weights->type);
```

Is it ok to skip the check?

Reply:
I've referenced TF2.8.0. which version did you refer?

Reply:
I'm seeing v2.10.0.

GitLens shows that the type checks were added about 2 years ago while the rest of the codes are written in 5 years ago.